nixos/modules/services/desktops/geoclue2.nix

```diff
@@ -61,6 +88,106 @@ in
         wantedBy = [ "default.target" ];
       };
     };
+
+    environment.etc."geoclue/geoclue.conf".text = ''
+    # Configuration file for Geoclue
```

Comment:
Do we need most of these options or can we leave them implicit?

Reply:
I'm seeing other projects just leaving it as is aside from using a custom geolocation api key from google.

Reply:
One concern with the explicit defaults is that user cannot override them. If they cannot be omitted, we would need to make them configurable.

Reply:
Very true, maybe I'll make an option only to create more application configurations (since that's the only thing I'll need currently), and remove `extraConfig`. I'll deal with the rest later.
Though I'm not sure of additional useful geoclue configuration.

Reply:
It is mainly about the things already in the text. For example, someone might want to switch to google services or other provider. Or change their leaderboard name.

Reply:
Yeah I also think this config file should be as minimal as possible, because it's not really possible to change anything in it, only add to it. The default upstream file is intended to be modified, which we can't like this

Alternatively, create a `config` option that transforms a Nix value into the config file syntax, and provide upstream defaults via `config.gnome-color-panel.allowed = mkDefault true;`. Then it's possible to override everything. I'm doing pretty much the same for https://github.com/NixOS/nixpkgs/pull/41467 (which I still need to finish)